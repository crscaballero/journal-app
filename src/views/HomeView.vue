<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">

    <h1>Hello World</h1>

    <button @click="goToDayBook" type="button" class="btn btn-primary">Primary</button>
    <button @click="goToDayBook" type="button" class="btn btn-secondary">Secondary</button>
    <button @click="goToDayBook" type="button" class="btn btn-success">Success</button>
  </div>
</template>

<script>
export default {
  methods: {
    goToDayBook() {
      this.$router.push({name: 'no-entry'})
    }
  }
}
</script>
