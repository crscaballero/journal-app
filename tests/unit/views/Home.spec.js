import { shallowMount } from '@vue/test-utils';

import HomeView from '@/views/HomeView.vue';

describe('HomeView.vue', () => {
	test('It must render the view correctly', () => {
		const wrapper = shallowMount(HomeView);
		expect(wrapper.html()).toMatchSnapshot();
	});
	test('Click in a button must redirect to "no-entry"', () => {
		const mockRouter = {
			push: jest.fn()
		}
		const wrapper = shallowMount(HomeView, {
			global: {
				mocks: {
					$router: mockRouter
				}
			}
		});
		wrapper.find('button').trigger('click');
		expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });
	});
});
