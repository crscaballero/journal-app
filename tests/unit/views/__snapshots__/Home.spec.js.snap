// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView.vue It must render the view correctly 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hello World</h1><button type="button" class="btn btn-primary">Primary</button><button type="button" class="btn btn-secondary">Secondary</button><button type="button" class="btn btn-success">Success</button>
</div>
`;
